candump.c

```diff
@@ -111,18 +111,18 @@ const char anichar[MAXANI] = {'|', '/', '-', '\\'};
 const char extra_m_info[4][4] = {"- -", "B -", "- E", "B E"};
 
 #define MAXLOGNAMESZ 100
-FILE *logfile = NULL;
+static FILE *logfile = NULL;
 static char log_filename[MAXLOGNAMESZ];
 
-unsigned char silent = SILENT_INI;
+static unsigned char silent = SILENT_INI;
 
 extern int optind, opterr, optopt;
 
 static volatile int running = 1;
-static volatile int flag_reopen_file = 0;
-static volatile unsigned long sighup_count = 0;
-int is_auto_log_name = 0;
-int open_log_file(const char * filename);
+static volatile int flag_reopen_file;
+static volatile unsigned long sighup_count;
+
+static int is_auto_log_name = 0;
```

Comment:
no need to initialize static vars with 0

Reply:
There are general issues with this patch. See https://github.com/linux-can/can-utils/pull/268#issuecomment-739410309
It does not make sense to discuss style problems until the general approach and concept is under discussion.

Reply:
> IMO if we implement this kind of multi-chunk file generation we should support either the size based triggers as suggested by @BugLight AND the SIGHUP approach from you.

What do you mean: "either...or" (as in one or the other) or "and" (as in both)?

Once file naming and new file generation is solved, triggering the switch to a new file due to `SIGHUP` or size limit is hopefully not that complicated.

Reply:
Yes, this was my intention!
But the way the filename is generated still needs some discussion IMO.
And finally I thought the PR https://github.com/linux-can/can-utils/pull/189 to be a simpler starting point, from which we might add the SIGHUP handling. But that's probably just my view.